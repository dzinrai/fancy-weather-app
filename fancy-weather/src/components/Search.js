import React, { useState } from 'react';
//import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import Button from './Button';

function Search(props) {
	const [search, setSearch] = useState('');
	function handleSubmit(event) {
		event.preventDefault();
		const searchCity = search;
		props.startSearch(searchCity);
	}
	return (
		<div className='search__container'>
			<form onSubmit={handleSubmit} className='search__form'>
				<input 
					className={(props.error ? 'wrong__input' : '')} 
					placeholder={props.text}
					onChange={(e) => setSearch(e.target.value)}
					value={search}
				/>
				<Button text={props.btnText} type='submit' value='Submit' />
			</form>
		</div>
	);

}

export default Search;