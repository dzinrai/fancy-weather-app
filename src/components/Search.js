import React, { useState, useEffect } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import Button from './Button';

function Search(props) {
	const [search, setSearch] = useState('');
	function handleSubmit(event) {
		event.preventDefault();
		props.startSearch(search);
	}
	return (
		<div>
			<form onSubmit={handleSubmit}>
				<input 
					className={props.classes} 
					placeholder={props.text}
					onChange={(e) => setSearch(e.target.value)}
				/>
				<Button text={props.btnText} classes='' type='submit' value='Submit' />
			</form>
		</div>
	);

}

export default Search;