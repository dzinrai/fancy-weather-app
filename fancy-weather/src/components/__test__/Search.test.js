import React from 'react';
import {cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Search from '../Search';

afterEach(cleanup);

test('proper search container', () => {
    const tree = renderer.create(<Search 
        text={'Search city'}
        btnText={'Search'}
        error={false}
        listening={true}
        supported={true}
        hideTips={true}
    />).toJSON();
    expect(tree).toMatchSnapshot();
})

test('proper search container', () => {
    const tree = renderer.create(<Search 
        text={'Search city'}
        btnText={'Search'}
        error={true}
        listening={true}
        supported={true}
        hideTips={true}
    />).toJSON();
    expect(tree).toMatchSnapshot();
})

test('proper search container', () => {
    const tree = renderer.create(<Search 
        text={'Search city'}
        btnText={'Search'}
        error={true}
        listening={false}
        supported={true}
        hideTips={true}
    />).toJSON();
    expect(tree).toMatchSnapshot();
})

test('proper search container', () => {
    const tree = renderer.create(<Search 
        text={'Search city'}
        btnText={'Search'}
        error={false}
        listening={false}
        supported={false}
        hideTips={true}
    />).toJSON();
    expect(tree).toMatchSnapshot();
})
