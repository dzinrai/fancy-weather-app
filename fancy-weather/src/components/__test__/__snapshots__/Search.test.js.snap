// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`proper search container 1`] = `
<div
  className="search__container"
>
  <form
    className="search__form"
    onSubmit={[Function]}
  >
    <input
      className=""
      onChange={[Function]}
      placeholder="Search city"
      value=""
    />
    <span
      className="mic"
      data-for="voice-search"
      data-tip={true}
      type="btn"
    />
    <button
      data-for={null}
      data-tip={false}
      type="submit"
      value="Submit"
    >
      Search
    </button>
  </form>
</div>
`;

exports[`proper search container 2`] = `
<div
  className="search__container"
>
  <form
    className="search__form"
    onSubmit={[Function]}
  >
    <input
      className="wrong__input"
      onChange={[Function]}
      placeholder="Search city"
      value=""
    />
    <span
      className="mic"
      data-for="voice-search"
      data-tip={true}
      type="btn"
    />
    <button
      data-for={null}
      data-tip={false}
      type="submit"
      value="Submit"
    >
      Search
    </button>
  </form>
</div>
`;

exports[`proper search container 3`] = `
<div
  className="search__container"
>
  <form
    className="search__form"
    onSubmit={[Function]}
  >
    <input
      className="wrong__input"
      onChange={[Function]}
      placeholder="Search city"
      value=""
    />
    <span
      className="mic"
      data-for="voice-search"
      data-tip={true}
      type="btn"
    />
    <button
      data-for={null}
      data-tip={false}
      type="submit"
      value="Submit"
    >
      Search
    </button>
  </form>
</div>
`;

exports[`proper search container 4`] = `
<div
  className="search__container"
>
  <form
    className="search__form"
    onSubmit={[Function]}
  >
    <input
      className=""
      onChange={[Function]}
      placeholder="Search city"
      value=""
    />
    <span
      className="mic"
      data-for="voice-search"
      data-tip={true}
      type="btn"
    />
    <button
      data-for={null}
      data-tip={false}
      type="submit"
      value="Submit"
    >
      Search
    </button>
  </form>
</div>
`;
